src/libraries/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -18,6 +18,25 @@ public static partial class Environment
         /// </summary>
         internal static bool IsSingleProcessor => ProcessorCount == 1;
 
+        private static bool s_privilegedProcess;
+        private static volatile int s_pendingPrivilegedProcess;
```

Comment:
This can be `bool`.

Alternatively, this can just a single static variable: 
```
private static volatile sbyte s_privilegedProcess;

...

s_privilegedProcess = IsAdminProcess() ? 1 : -1;
...
return s_privilegedProcess == 1;
```